File: src/main/java/me/alegian/thavma/impl/init/registries/deferred/T7EntityTypes.java
package me.alegian.thavma.impl.init.registries.deferred;

import me.alegian.thavma.impl.Thavma;
import me.alegian.thavma.impl.common.entity.VisEntity;
import net.minecraft.core.registries.Registries;
import net.minecraft.world.entity.EntityType;
import net.minecraft.world.entity.MobCategory;
import net.minecraft.world.entity.item.ItemEntity;
import net.neoforged.neoforge.registries.DeferredHolder;
import net.neoforged.neoforge.registries.DeferredRegister;

public class T7EntityTypes {
  public static final DeferredRegister<EntityType<?>> REGISTRAR = DeferredRegister.create(Registries.ENTITY_TYPE, Thavma.MODID);

  public static final DeferredHolder<EntityType<?>, EntityType<ItemEntity>> FANCY_ITEM = REGISTRAR.register("fancy_item", () -> EntityType.Builder.<ItemEntity>of(ItemEntity::new, MobCategory.MISC).sized(0.25F, 0.25F).eyeHeight(0.2125F).clientTrackingRange(6).updateInterval(20).build("fancy_item"));
  public static final DeferredHolder<EntityType<?>, EntityType<VisEntity>> VIS = REGISTRAR.register("vis", () -> EntityType.Builder.<VisEntity>of((entityType, level) -> new VisEntity(level, null), MobCategory.MISC).sized(0.25F, 0.25F).eyeHeight(0.2125F).clientTrackingRange(6).updateInterval(20).build("vis"));
}

File: src/main/java/me/alegian/thavma/impl/init/registries/deferred/T7EntityTypes.kt
package me.alegian.thavma.impl.init.registries.deferred

import me.alegian.thavma.impl.Thavma
import me.alegian.thavma.impl.common.entity.AngryZombieEntity
import me.alegian.thavma.impl.common.entity.VisEntity
import net.minecraft.core.registries.Registries
import net.minecraft.world.entity.EntityType
import net.minecraft.world.entity.MobCategory
import net.minecraft.world.entity.item.ItemEntity
import net.neoforged.neoforge.registries.DeferredRegister

object T7EntityTypes {
    val REGISTRAR = DeferredRegister.create(Registries.ENTITY_TYPE, Thavma.MODID)

    val FANCY_ITEM = REGISTRAR.register("fancy_item") { ->
        EntityType.Builder.of({ entityType, level -> ItemEntity(entityType, level) }, MobCategory.MISC)
            .sized(0.25f, 0.25f)
            .eyeHeight(0.2125f)
            .clientTrackingRange(6)
            .updateInterval(20)
            .build("fancy_item")
    }

    val VIS = REGISTRAR.register("vis") { ->
        EntityType.Builder.of({ _, level -> VisEntity(level, null) }, MobCategory.MISC)
            .sized(0.25f, 0.25f)
            .eyeHeight(0.2125f)
            .clientTrackingRange(6)
            .updateInterval(20)
            .build("vis")
    }

    val ANGRY_ZOMBIE = REGISTRAR.register("angry_zombie") { ->
        EntityType.Builder.of({ entityType, level -> AngryZombieEntity(entityType, level) }, MobCategory.MONSTER)
            .sized(0.6F, 1.95F)
            .eyeHeight(1.74F)
            .passengerAttachments(2.0125F)
            .ridingOffset(-0.7F)
            .clientTrackingRange(8)
            .build("angry_zombie")
    }
}
